### PR TITLE
ipatests: Added testcase to check that ipa-healthcheck tool displays warning if no range is set

### PR DESCRIPTION
**Pull request by menonsudhir**
ipatests: Added testcase to check that ipa-healthcheck tool displays warning if no range is set.
Issue: https://github.com/freeipa/freeipa-healthcheck/issues/60

### CODE REVIEW

**Comment by rcritten** [comment]
./ipatests/test_integration/test_ipahealthcheck.py:46:80: E501 line too long (85 > 79 characters)
./ipatests/test_integration/test_ipahealthcheck.py:49:80: E501 line too long (99 > 79 characters)
./ipatests/test_integration/test_ipahealthcheck.py:50:28: E128 continuation line under-indented for visual indent
./ipatests/test_integration/test_ipahealthcheck.py:56:80: E501 line too long (86 > 79 characters)
./ipatests/test_integration/test_ipahealthcheck.py:65:17: E113 unexpected indentation
./ipatests/test_integration/test_ipahealthcheck.py:79:42: E128 continuation line under-indented for visual indent

**Comment by rcritten** [comment]
The first line of a git commit should convey the basic jist of what the patch does in 60 characters or less. This is often pretty challenging.

I'd change the commit to:
```
ipatests: check that ipa-healthcheck warns if no dna range is set

Added testcase to verify that ipa-healthcheck tool displays a 
warning if no DNS range is set. It previously just reported at the
SUCCESS level that no range was set.

Issue: freeipa/freeipa-healthcheck#60
```

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 1]
```diff
@@ -413,6 +412,95 @@ def test_source_ipa_roles_check_crlmanager(self):
...
+                "define a range then users and groups cannot be created."
+            )
+
+        # Now kinit as admin and adduser on replica
+        tasks.kinit_admin(self.replicas[0])
+        ipauser = "ipauser2"
+        first = "ipauser"
+        last = "test"
+        password = "Secret123"
+        self.replicas[0].run_command(
```
Comment (fcami) [inline ipatests/test_integration/test_ipahealthcheck.py]: see tasks.py::user_add()

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 2]
```diff
@@ -413,6 +412,95 @@ def test_source_ipa_roles_check_crlmanager(self):
...
+                "IPADNARangeCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
```
Comment (fcami) [inline ipatests/test_integration/test_ipahealthcheck.py]: This needs to be factored in a function that takes as argument the json output and the check you want to check. This way you won't have duplicate code.

**Comment by rcritten** [comment]
I fixed a few things raised, working on abstracting the healthcheck calls themselves (I pushed my suggestion patch a bit prematurely).

**Comment by fcami** [comment]
LGTM. Please remove the temp commit and I'll ack.

**Comment by rcritten** [comment]
I want to see what @menonsudhir wants to do with my suggestions patch. It removes a bunch of unused variables and some cleanups unrelated to the change.

**Comment by menonsudhir** [comment]
@rcritten whatever changes were done in this PR LGTM, ACK

**Comment by rcritten** [comment]
master:

* 4e3a2bd6bff79c17bc439555dcf41a11d0569e6f ipatests: check that ipa-healthcheck warns if no dna range is set
* 44e73428441a070f26a791ee2816f2989713ba8d Move execution of ipa-healthcheck to a separate function

**Pull request closed without merging**